### ISSUE TITLE
[Feature] sort repository 

### ISSUE THREAD

**Issue by TimerWolf** [go-gitea/gitea#25586]
### Feature Description

Why is it not possible to sort the order of the information inside a repository, after all it is a list ?

It would be nice if "files / folders" could be sorted after "files / folders" maybe after "name" or the commits could be sorted after "name" or why not the "commits dates" be sorted after "first / last" commit ?

I do not see why this not possible, and not implanted as it should be in my eyes very neat and obvious function.

And know i am not sure how it is sorted by default, but as well maybe have a setting under the user account that lets you set the "default" order it should be sorted in, if it are after ("file/folder name" / commit / date) as an example

**Comment by silverwind** [comment]
I don't see much use of sorting the files table and to make it sortable, some headers would likely needed to be added which imho bloats the UI. Sort by commit name is useless. Sort by commit date is semi-useful, but I think more often than not people are looking for a file than for commit dates.

**Comment by TimerWolf** [comment]
It was just an example, what i was after was to sort after name but while you add the "sorting function" if it would be added it never bad to have some extra features if anyone would like them.....

**Comment by mpeter50** [comment]
You mean sorting the files of a repository, right? Isn't it already sorted by name?

**Comment by TimerWolf** [comment]
Not if you look at my repo on this image, maybe it is a bug?

![alt text](https://i.imgur.com/aK7iOvI.png)

**Comment by mpeter50** [comment]
That is interesting. Maybe the devs can shed some light on it.

What I do know is that in most cases, it looks to be sorted by filename, whether it is a directory, and wthether it is hidden (starts with a dot):
- https://gitea.com/gitea/gitea_mirror
- https://gitea.com/gitea/tea
- https://codeberg.org/jannis/FitoTrack/
- https://codeberg.org/Freeyourgadget/Gadgetbridge

Also, I just noticed that names starting with uppercase letters are sorted earlier, too.
From you Ctrl+F-ing it looks they all start with the same characters until the number. Maybe there are inivisble characters before or after some of the names highlighted part, or some of the numbers are weird alternatives of the actual numbers?

Maybe you could look in the browser developer tools to see if theres an invisible difference thats shown here.

**Comment by delvh** [comment]
If I'm not mistaken, the sort order is the one Git returns.
And Git should return the files in alphabetic order by default.
The fact that this is not the case for you indicates one of two things for me:
1. Your `v8` filename contains an invisible space for some reason (i.e. a zero-width-space)
2. Git has a bug…

I don't see many other options right now.

**Comment by TimerWolf** [comment]
Maybe that´s the problem, this "git directory" is created in "gitea" and do not have any "offline" repository.

**Comment by delvh** [comment]
I am quite confused by what you mean.
Gitea is in this regard simply a proxy for Git:
We simply query all file and commit-related information from Git.
If it it doesn't exist for Git, it doesn't exist for Gitea.

**Comment by TimerWolf** [comment]
Exactly like that but i did try to rename it and it did work....

I will keep my eyes open as i do not think it was any "hidden/wrong" char in it before but you should never say never...

**Comment by danielwagn3r** [comment]
Seem's to be a duplicate of #19521

**Comment by danielwagn3r** [comment]
Fair enough. I saw a relation because both are lists which sorting can't be controlled by the user currently. Depending on the underlying components the resolution could be shared among both issues. I.e. a custom sortable table control - one containing repsitory contents (this issue) the other container commit history (the other issue).

**Comment by netdiver** [comment]
I would like to add my vote to this feature. We are using Git repos to track down changes to network devices configuration (with Oxidized) and using Gitea to have a GUI for repo browsing. It would be useful for us to order by date of commits, to get, with a quick look, which device has been modified recently.
